core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.spec.ts

```diff
@@ -269,4 +263,34 @@ describe('Exploration save and publish buttons component', function() {
     spyOn(editabilityService, 'isTranslatable').and.returnValue(false);
     expect($scope.isEditableOutsideTutorialMode()).toBe(false);
   });
+
+  it('should display publish button', function() {
+    $scope.explorationCanBePublished = false;
+
+    explorationRightsService.onExplorationUnpublished.emit();
+    $scope.$apply();
```

Comment:
Why do we need this $apply call?

Reply:
yes

Reply:
I didn't get it, I mean can you explain why we need this $apply call?

Reply:
https://stackoverflow.com/a/40100803/15100413

Looks like in jasmine testing the `$scope.$apply` is not called automatically

https://stackoverflow.com/a/32786744/15100413